packages/dumb/src/provider/trends.provider.hook.tsx

```diff
@@ -28,20 +28,30 @@
 import { useState } from 'react'
 import { type TrendsDisplayFlags } from '../models/trends-display-flags.model'
 import { DisplayFlag } from '../models/enums/display-flag.enum'
+import { type CbgDateTrace } from '../models/cbg-date-trace.model'
+import { type CbgPositionData } from '../models/cbg-position-data.model'
+import { type FocusedCbgDateTrace } from '../models/focused-cbg-date-trace.model'
 
 export interface TrendsContextResult {
   displayFlags: TrendsDisplayFlags
+  focusCbgDateTrace: (cbgDateTrace: CbgDateTrace, position: CbgPositionData) => void
+  focusedCbgDateTrace: FocusedCbgDateTrace
   toggleCbgSegments: (displayFlag: DisplayFlag) => void
+  unfocusCbgDateTrace: () => void
 }
 
 export const useTrendsProviderHook = (): TrendsContextResult => {
+  const emptyCbgDateTraceValue = undefined as unknown as FocusedCbgDateTrace
+
   const [displayFlags, setDisplayFlags] = useState<TrendsDisplayFlags>({
     cbg50Enabled: true,
     cbg80Enabled: true,
     cbg100Enabled: true,
     cbgMedianEnabled: true
   })
 
+  const [focusedCbgDateTrace, setFocusedCbgDateTrace] = useState<FocusedCbgDateTrace>(emptyCbgDateTraceValue)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏You are tricking typescript in believing that the value is always a `FocusedCbgDateTrace`, but the value is undefined. Why not be honest with him ans have the useState being typed as `FocusedCbgDateTrace | undefined`? This seems cleaner to me.